app/src/apis/memberApi.ts

```diff
@@ -30,12 +31,13 @@ export class FetchMemberAPI implements MemberAPI {
     const response = await fetch(`${BACKEND_URL}/api/member/${id}`, requestOptions)
     const resObj = await response.json()
     if (!resObj.success) {
+      await AuthAPI.verifyToken(token)
       throw new Error(resObj.message)
     }
     return resObj.members
```

Comment:
maybe for the next refactor: 

i think we can create something like composable for fetching data from API so we dont have to write verifyToken in each function where we fetch data from API, in the composable there is a check if the route is public we dont have to verify the token, and if it's protected route we need to verify the token before we fetch the data.

Reply:
We're not verifying it each time. Only when there is an unauthorized response from the backend we call the verifyToken method.

Reply:
right, but we still have to define it in every function that will call our API, so we can implement DRY principle https://en.wikipedia.org/wiki/Don%27t_repeat_yourself


```typescript
export function useFetch(url: string) {
  // some other codes
  function fetchApi(url: string) {
    const res = await fetch(url)
    if (res.statusCode == 401) throw Unauthorize
  
    return res
  }
    // some other codes
}
```

```typescript
function findAllMember() {
  const { fetchApi } = useFetch()

  const res = await fetchApi(url)
  // no longer need to add check if it's unauthorize

  // rest of the code..
}

function findAllUser() {
  const { fetchApi } = useFetch()

  const res = await fetchApi(url)
  // no longer need to add check if it's unauthorize

  // rest of the code..
}
```

or u can use VueUse https://vueuse.org/core/useFetch/ but im not really sure about it

Reply:
Wow that sounds neat. I'll create a separate issue for this refactor.

Reply:
But I don't think a composable is appropriate in this case. There's no state that needs to be maintained. Based on this https://www.youtube.com/watch?v=N0QrFKBZuqA , this does not satisfy any of the below conditions for it to be a composable
- Your function uses other composable 
- Your function uses lifecycle hooks
- Your function has stateful logic - ref

Reply:
we can add state like isFetching, isSuccess, error, data like in the useFetch in vueUse